proxy/director.go

```diff
@@ -65,6 +66,13 @@ func (d *director) refresh() {
 		}
 		endpoints = append(endpoints, newEndpoint(*uu))
 	}
+
+	// shuffle array to avoid connections being "stuck" to a single endpoint
+	for i := range endpoints {
```

Comment:
a simple shuffling is good enough for now. If we want the proxy to coalesce watches, we need to be more careful with this.